packages/medusa/src/api/routes/store/payment-collections/manage-payment-session.ts

```diff
@@ -0,0 +1,97 @@
+import { IsString } from "class-validator"
+
+import { EntityManager } from "typeorm"
+import { PaymentCollectionService } from "../../../../services"
+
+/**
+ * @oas [post] /payment-collections/{id}/sessions
```

Comment:
If we send the provider_id at the URL I believe it won't be clear that only the last provider_id will be selected and other ones will be deleted/canceled. Or is the idea to keep the flow like carts where all payment sessions are created and the endpoint just set the flag `selected` in one of them?

People might think they can control multiple sessions with multiple providers by calling this endpoint using different provider_ids.
Maybe we can rename `/sessions` but I believe the provider_id should be passed as the payload.
But I don't have a strong opinion on it. I am ok with other approaches.

Reply:
> Or is the idea to keep the flow like carts where all payment sessions are created and the endpoint just sets the flag selected in one of them

Good question. I believe it's a tradeoff. Keeping the sessions would lead to a more performant endpoint when the customer is shuffling between the session, but deleting/canceling them would clean up the DB and eventually free up space.

I am fine keeping it as is, and then we can reevaluate based on feedback from users.